### ISSUE TITLE
Unable to Reproduce Paper's Results on DVF Raw Dataset (Inference Only, No Fine-Tuning) / AUC Does Not Exceed 75%

### ISSUE THREAD

**Issue by saimeur** [SparkleXFantasy/MM-Det#9]
Hi,

First of all, thank you for your work. Reading your paper was truly insightful.

I am currently trying to reproduce your results on the DVF dataset (Table 1), using the raw data downloaded from the Google Drive link you provided. I followed the steps outlined in this section: [insert link](https://github.com/SparkleXFantasy/MM-Det/tree/main/dataset). Specifically, I:
- Downloaded the weights from this link: [https://drive.google.com/drive/folders/1RRNS8F7ETZWrcBu8fvB3pM9qHbmSEEzy]
- Used `prepare_reconstructed_dataset.py` to convert videos into frame sequences and generate reconstructed frames.
- Used `prepare_mm_representations.py` to cache the MMFR.

Here is the command I used:
```bash
python test.py \
  --classes videocrafter1 zeroscope opensora sora pika stablediffusion stablevideo \
  --ckpt ./weights/MM-Det/current_model.pth \
  --data-root $RECONSTRUCTION_DATASET_ROOT \
  --cache-mm \
  --mm-root $MM_REPRESENTATION_ROOT \
  --sample-size -1
```

I made a small modification to the code to also output accuracy, F1 score, and confusion matrix alongside the AUC.

Here are my results for the "opensora" subset:
```
Mon Dec 23 02:59:52 2024       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 555.42.06              Driver Version: 555.42.06      CUDA Version: 12.5     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA L40S                    Off |   00000000:64:00.0 Off |                    0 |
| N/A   30C    P8             33W /  350W |       1MiB /  46068MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
Conda environment activated

CUDA version:
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Sun_Mar_21_19:15:46_PDT_2021
Cuda compilation tools, release 11.3, V11.3.58
Build cuda_11.3.r11.3/compiler.29745058_0

{'data_root': 'paper_outputs/reconstruction_dataset', 'ckpt': './weights/MM-Det/current_model.pth', 'lmm_ckpt': 'sparklexfantasy/llava-7b-1.5-rfrd', 'lmm_base': None, 'st_ckpt': 'weights/ViT/vit_base_r50_s16_224.orig_in21k/jx_vit_base_resnet50_224_in21k-6f7c7740.pth', 'st_pretrained': True, 'model_name': 'MMDet', 'expt': 'MMDet_01', 'window_size': 10, 'conv_mode': 'llava_v1', 'new_tokens': 64, 'selected_layers': [-1], 'interval': 200, 'load_8bit': False, 'load_4bit': False, 'seed': 42, 'gpus': 1, 'cache_mm': True, 'mm_root': 'paper_outputs/mm_representation_recons', 'debug': False, 'classes': ['opensora'], 'bs': 1, 'mode': 'test', 'sample_size': -1, 'num_workers': 1}
Validation on ['opensora'].
 ===== test_dataset_configs ===== 
[{'videocrafter1': {'data_root': 'paper_outputs/reconstruction_dataset/videocrafter1', 'dataset_type': 'VideoFolderDatasetForReconsWithFn', 'mode': 'test', 'selected_cls_labels': [('0_real', 0)], 'sample_method': 'entire'}, 'zeroscope': {'data_root': 'paper_outputs/reconstruction_dataset/zeroscope', 'dataset_type': 'VideoFolderDatasetForReconsWithFn', 'mode': 'test', 'selected_cls_labels': [('0_real', 0)], 'sample_method': 'entire'}, 'opensora': {'data_root': 'paper_outputs/reconstruction_dataset/opensora', 'dataset_type': 'VideoFolderDatasetForReconsWithFn', 'mode': 'test', 'selected_cls_labels': [('0_real', 0), ('1_fake', 1)], 'sample_method': 'entire'}}]
Load checkpoint from ./weights/MM-Det/current_model.pth
 ============================== 
opensora
AUC: 0.7312017142857143
ACC: 0.24711111111111111
F1: 0.3711952487008166
Confusion Matrix: [[56, 1694], [0, 500]]

```

It seems the model has a significant bias towards classifying almost every video as fake (~75% of my dataset is classified as false positives). I may have missed something in my implementation or perhaps I'm using untrained weights.

For all subsets (opensora, pika, etc.), the AUC does not exceed 75%, which is far from the results reported in your paper.

Do you have any insights on what might be going wrong?

**Comment by lds217** [comment]
Hi Saimeur,

I hope your issue can be solved. Off topic, can I contact you and have the fixed code you are using? Since I know the straightaway code pulled from GitHub is not correct 100%. We are doing research related to this topic and need this to run. We had it running a little bit, but the AUC is way off – even worse than yours (we modified some of the code).

Thank you for your help!

**Comment by SparkleXFantasy** [comment]
Hi. Thank you for your attention. We have provided our pre-cached reconstruction dataset and MM representations in the repository. Our environment is pytorch 2.2.1+cu121. I wonder if you could compare your cached file with our MM representations to verify whether the results are consistent.

**Comment by saimeur** [comment]
> Hi Saimeur,
> 
> I hope your issue can be solved. Off topic, can I contact you and have the fixed code you are using? Since I know the straightaway code pulled from GitHub is not correct 100%. We are doing research related to this topic and need this to run. We had it running a little bit, but the AUC is way off – even worse than yours (we modified some of the code).
> 
> Thank you for your help!

Hi @lds217, I made some changes to get it running on my PC, but the results are really bad. I don't think I have a "fixed" version of the original code. Plus, my coding skills aren't great. If you want, I can try to clean up my code a bit and send you a zip file.


> Hi. Thank you for your attention. We have provided our pre-cached reconstruction dataset and MM representations in the repository. Our environment is pytorch 2.2.1+cu121. I wonder if you could compare your cached file with our MM representations to verify whether the results are consistent.

Thanks for your response, @SparkleXFantasy. I'm not familiar with ByteDance, and I found it easier to download the raw DVF dataset using gdown. This method allows me to directly transfer the dataset from Google Drive to my server, where I store all my data. I would like to download the pre-cached reconstruction dataset and MM representations, unfortunately downloading the data requires the Netdisk client, using gdown simplifies the process. If you could provide a Google Drive link with the MM representations and the pre-cached reconstruction, it would be incredibly helpful.

I noticed that you recently modified the code. Do I need to fork the repository to address the discrepancies in my results? I'm only interested in inference, not training the model. I'm currently working with a Git clone from December 17th.

**Comment by lds217** [comment]
> Hi @lds217, I made some changes to get it running on my PC, but the results are really bad. I don't think I have a "fixed" version of the original code. Plus, my coding skills aren't great. If you want, I can try to clean up my code a bit and send you a zip file.

My apologies for the wording earlier—the project and paper are highly insightful, and "fixed" was not the right term. I believe we made similar minor adjustments to the config to get it running as well.

Yes, I’d appreciate it if you could share the zip file.

Thank you!

**Comment by SparkleXFantasy** [comment]
Hi, @saimeur . Thank you for your attention. We have recently updated a tiny version of DVF as well as the cached reconstruction dataset and MMFR [here](https://github.com/SparkleXFantasy/MM-Det?tab=readme-ov-file#diffusion-video-forensics-dataset), where each class in the dataset contains 10 short videos. Noticing that you have already cached these data, we'd apprecicate it much if you could check your representations with ours to check whether there is any discrepency.

**Comment by saimeur** [comment]
Sorry for the misclick, I didn't mean to close this issue 😅

**Comment by saimeur** [comment]
Sorry for the misclick, I didn't mean to close this issue 😅

**Comment by SparkleXFantasy** [comment]
Hi, @saimeur 
We get the following performance on AUC based on the tiny version of DVF.
|Videocrafter1| Zeroscope|OpenSora|Sora|Pika|Stable Diffusion | Stable Video|
| --- | --- | --- | --- | --- | --- | --- |
| 90.99 | 92.49 | 90.25 | 90.99 | 90.75 | 89.00 | 84.25 |
